src/workerd/api/sql-test.js

```diff
@@ -322,6 +322,61 @@ async function test(sql) {
   assert.equal(sql.voluntarySizeLimit, 1073741823 * 4096);
   sql.voluntarySizeLimit = 65536;
   assert.equal(sql.voluntarySizeLimit, 65536);
+
+  storage.put("txnTest", 0);
+
+  // Try a transaction while no implicit transaction is open.
+  await scheduler.wait(1);  // finish implicit txn
+  await storage.transaction(async () => {
+    storage.put("txnTest", 1);
+    assert.equal(await storage.get("txnTest"), 1);
+  });
```

Comment:
Will the call to `storage.transaction` return the value returned by the callback? If so, it would be good to test that here with a slight modification:

```
assert.equal(await storage.transaction(async () => {
  storage.put("txnTest", 1);
  return storage.get("txnTest");
}), 1);
```